modules/core/shared/src/main/scala/lucuma/core/validation/InputValidWedge.scala

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.validation
+
+import cats.data.NonEmptyChain
+import cats.syntax.all._
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.Positive
+import eu.timepit.refined.types.numeric.PosBigDecimal
+import eu.timepit.refined.types.string.NonEmptyString
+import lucuma.core.optics._
+import monocle.Iso
+import monocle.Prism
+import singleton.ops._
+
+import scala.annotation.unused
+
+/**
+ * Convenience version of `ValidWedge` when the error type is `NonEmptyChain[NonEmptyString]` and
+ * `T` is `String`.
+ */
+object InputValidWedge {
+
+  /**
+   * Build an `InputValidWedge` that's always valid and doesn't normalize or format
+   */
+  val id: InputValidWedge[String] = ValidWedge.id
+
+  /**
+   * Build an `InputValidWedge` from `getValid` and `reverseGet` functions
+   */
+  def apply[A](
+    getValid:   String => EitherErrors[A],
+    reverseGet: A => String
+  ): InputValidWedge[A] =
+    ValidWedge(getValid, reverseGet)
+
+  /**
+   * Build an `InputValidWedge` from a `Format`
+   */
+  def fromFormat[A](
+    format:       Format[String, A],
+    errorMessage: NonEmptyString = "Invalid format"
+  ): InputValidWedge[A] =
+    ValidWedge(
+      format.getOption.andThen(_.toRight(errorMessage).toEitherErrors),
+      format.reverseGet
+    )
+
+  /**
+   * Build an `InputValidWedge` from a `Prism`
+   */
+  def fromPrism[A](
+    prism:        Prism[String, A],
+    errorMessage: NonEmptyString = "Invalid value"
+  ): InputValidWedge[A] =
+    fromFormat(Format.fromPrism(prism), errorMessage)
+
+  /**
+   * Build an `InputValidWedge` from an `Iso`
+   */
+  def fromIso[A](iso: Iso[String, A]): InputValidWedge[A] =
+    ValidWedge(
+      (iso.get _).andThen(_.asRight),
+      iso.reverseGet
+    )
+
+  /**
+   * Build a `InputValidWedge` for `BigDecimal` truncated to `Dec` decimals.
+   */
+  def truncatedBigDecimal[Dec <: XInt](implicit
+    @unused req: Require[&&[Dec >= 0, Dec < 10]],
+    vo:          ValueOf[Dec]
```

Comment:
If we no longer have Truncated* types for BigDecimal, RA, Dec, etc., do we still need the decimals as a type parameter, or could we just pass in a NonNegInt as a regular paramter. My recollection is we needed it this way because the type of TruncatedBigDecimal depended on the number of decimals - which is not a thing anymore. The best way to deal with singleton ops for scala 3 would be to eliminate them...

The upper bound of 10 was just an arbitrary sanity thing. We could either leave that to programmer discretion, or have the parameter be a refined Interval.

Reply:
I think you are right, we could just make the decimals a regular parameter.